database/ffldb/db.go

```diff
@@ -1669,6 +1669,94 @@ func (tx *transaction) writePendingAndCommit() error {
 	return tx.db.cache.commitTx(tx)
 }
 
+// PruneBlocks deletes the block files until it reaches the target size
+// (specificed in bytes).  Throws an error if the target size is below
+// the maximum size of a single block file.
+//
+// This function is part of the database.Tx interface implementation.
+func (tx *transaction) PruneBlocks(targetSize uint32) ([]chainhash.Hash, error) {
+	// Ensure transaction state is valid.
+	if err := tx.checkClosed(); err != nil {
+		return nil, err
+	}
+
+	// Ensure the transaction is writable.
+	if !tx.writable {
+		str := "prune blocks requires a writable database transaction"
+		return nil, makeDbErr(database.ErrTxNotWritable, str, nil)
+	}
+
+	if targetSize < tx.db.store.maxBlockFileSize {
+		return nil, fmt.Errorf("got target size of %d but it must be greater "+
+			"than %d, the max size of a single block file",
+			targetSize, maxBlockFileSize)
+	}
+
+	first, last, lastFileSize, err := scanBlockFiles(tx.db.store.basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// If we have no files on disk or just a single file on disk, return early.
+	if first == last {
+		return nil, err
+	}
+
+	// Last file number minus the first file number gives us the count of files
+	// on disk minus 1.  We don't want to count the last file since we can't assume
+	// that it is of max size.
+	maxSizeFileCount := last - first
+
+	// If the total size of block files are under the target, return early and
+	// don't prune.
+	totalSize := lastFileSize + (tx.db.store.maxBlockFileSize * uint32(maxSizeFileCount))
+	if totalSize <= targetSize {
+		return nil, err
+	}
+
+	log.Tracef("Using %d more bytes than the target of %d MiB. Pruning files...",
+		totalSize-targetSize,
+		targetSize/(1024*1024))
+
+	deletedFiles := make(map[uint32]struct{})
+	for i := uint32(first); i <= uint32(last); i++ {
+		err = tx.db.store.deleteFileFunc(i)
+		if err != nil {
+			return nil, fmt.Errorf("PruneBlocks: Failed to delete block file "+
+				"number %d: %v", i, err)
+		}
+
+		deletedFiles[i] = struct{}{}
+
+		// We'll never delete the last block file the file size will always
+		// be the maxBlockFileSize.
+		totalSize -= tx.db.store.maxBlockFileSize
+
+		// If we're already at or below the target usage, break and don't
+		// try to delete more files.
+		if totalSize <= targetSize {
+			break
+		}
+	}
+
+	var deletedBlockHashes []chainhash.Hash
+	cursor := tx.blockIdxBucket.Cursor()
+	for ok := cursor.First(); ok; ok = cursor.Next() {
+		loc := deserializeBlockLoc(cursor.Value())
+
+		_, found := deletedFiles[loc.blockFileNum]
+		if found {
+			deletedBlockHashes = append(deletedBlockHashes, *(*chainhash.Hash)(cursor.Key()))
```

Comment:
Not quite sure what you mean here. Mine is set to 8 spaces already and I couldn't find examples of what you mentioned here.

Also, afaik, tabspace shouldn't matter with tabs anyways since the editor you're using should render it as whatever you set right?

Reply:
I mean that the code in this PR often flows over the targeted 80 character line length limit we use in `lnd`. Though I see that is not specifically mentioned in this project's code contribution guidelines. So feel free to ignore as much of the existing code also doesn't seem to use that limit.